web/src/components/dashboard/SidebarCardFundingNudge.tsx

```diff
@@ -13,16 +23,104 @@ type Props = {
 
 export const SidebarCardFundingNudge = (props: Props): ReactElement => {
   const [modalOpen, setModalOpen] = useState<boolean>(false);
+  const [profileData, setProfileData] = useState<ProfileData>(createEmptyProfileData());
+
+  const { queueUpdateTaskProgress } = useUpdateTaskProgress();
+  const close = (): void => {};
+  const onSuccess = (): void => {};
   const router = useRouter();
   const { updateQueue, business } = useUserData();
 
-  const updateToUpAndRunningAndCompleteTaxTask = async (): Promise<void> => {
+  useMountEffectWhenDefined(() => {
     if (!business) return;
+    setProfileData(business.profileData);
+  }, business);
+
+  const displayBusinessName = (): boolean => {
+    return LookupLegalStructureById(business?.profileData.legalStructureId).elementsToDisplay.has(
+      "businessName"
+    );
+  };
+
+  const displayResponsibleOwnerName = (): boolean => {
+    return LookupLegalStructureById(business?.profileData.legalStructureId).elementsToDisplay.has(
+      "responsibleOwnerName"
+    );
+  };
+
+  const updateToUpAndRunningAndCompleteTaxTask = async (): Promise<void> => {
+    if (!business || !updateQueue) return;
     await updateQueue
       ?.queueProfileData({
         operatingPhase: OperatingPhaseId.UP_AND_RUNNING,
       })
       .update();
+
+    const encryptedTaxId =
+      profileData.taxId === business.profileData.taxId ? profileData.encryptedTaxId : undefined;
```

Comment:
dust - I don't really understand what this is for? Especially since I think profileData and business.profileData should be the same.

Reply:
Attempts at refactoring this (i.e. -- moving this logic to a `lib` folder to reuse in `SidebarCardFundingNudge.tsx`) caused the jest tests to fail.